JULIERAJ/KIDS-FIRST-Front
# Fix/136/authen layout

##136 Fix: sign up/in layout

## Requirements
- When errors occur in sign up/in form, layout shouldn't drag and change.
- Add autocompleting to the email input, allow copying


## Note
Check in Chrome - there are differences in Safari, there will be a different ticket to handle this

## Test Steps

Requirement 1: When errors occur in the sign-up/sign-in form, the layout shouldn't drag and change.
Test Steps:

1. Load the Sign-Up Form:

Open the application and navigate to the sign-up form.

2. Trigger Errors:

Attempt to submit the form with empty required fields.
Enter invalid values (e.g., invalid email format, short password).

3. Observe Layout Stability:

Verify that error messages appear without changing the overall layout or shifting other elements.
Check that input fields, buttons, and other elements maintain their positions.

4. Validate Error Message Display:

Ensure that error messages are clearly visible and correctly positioned without overlapping other elements.

Expected Results:
The layout remains consistent and stable when errors are displayed.
Error messages appear without causing any shifts or dragging of other form elements.

Requirement 2: Add autocompleting to the email input, allowing copying.
Test Steps:

1. Load the Sign-Up Form:

Open the application and navigate to the sign-up form.

2. Check Autocomplete Functionality:

Start typing in the email input field.
Verify that the browser suggests previously entered email addresses (if any).

3. Copy Email Input:

Enter a valid email address in the email input field.
Select and copy the email address using the keyboard shortcut (Ctrl+C) or context menu.
Paste the copied email address into another field or document to ensure it was copied correctly.

4. Submit the Form:

Fill out the rest of the form with valid information and submit it.
Verify that the form submission works as expected with the auto-completed and copied email.

Expected Results:
The email input field should have autocomplete enabled, suggesting previously entered email addresses.
The email address should be selectable and copyable.
The form should submit successfully with an auto-completed and copied email address.

## UI/UX Outcome
- see attached videos
https://github.com/JULIERAJ/KIDS-FIRST-Front/assets/124208414/52828816-6e9c-4d87-b54d-f633b81442fc

https://github.com/JULIERAJ/KIDS-FIRST-Front/assets/124208414/6a64bf2b-8024-4795-9e16-36684b0fa015
